python/private/python.bzl

```diff
@@ -357,6 +357,13 @@ def _process_single_version_overrides(*, tag, _fail = fail, default):
             for platform in sha256
         }
 
+    if tag.flag_values:
+        # Normalize flag keys to strings
+        flag_values = {str(k): v for k, v in tag.flag_values.items()}
```

Comment:
I think the normalization should be done inside `python_register_toolchains` call, because that macro should work equally well with label keyed dicts and with string dicts.